### PR TITLE
Adding module Common/Types with ValuePack as first type

### PR DESCRIPTION
**Pull request by matthiasrichter**
ValuePack allows to assemble multiple values with a fixed layout
of bit widths in one type T. Name is tentative

@sawenzel this is the prototype we have been discussing today, I have added some more operators and also the field setter.

### CODE REVIEW

File: Common/Types/include/Types/ValuePack.h [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+};
+
+// specialization terminating the recursive meta function
+template<typename Rep, typename End>
+struct CodeInit<Rep, End, End> {
+  static Rep value() {return 0;}
+};
+
+/**
+* @class ValuePack
```
Comment (sawenzel) [inline Common/Types/include/Types/ValuePack.h]: I would add a somewhat longer comment explaining this class:
```
ValuePack is a generic template class packing/unpacking multiple integral values of given 
bit sizes into/from an underlying storage type Rep.
```

Comment (matthiasrichter) [reply Common/Types/include/Types/ValuePack.h]: good, will add this, anything else we should include in the first version?

**Comment by sawenzel** [comment]
@ktf: Well the main motivation for ValuePack is to have something **configurable** and **reusable** that hides the low level details. Implementing it in terms of a union seems possible (but I would exactly like to avoid that the user copy pastes such code in concrete applications).

**Comment by ktf** [comment]
To me what this does is to replace a named bitfield with an unnamed one, where parts are accessed in a positional way. How can that be better? Do you have a concrete example where this gives advantages?

**Pull request closed without merging**